### DISCUSSION TITLE
Predefined multi-lingual names for ingredients

### DISCUSSION THREAD

**Discussion by nununo** [mealie-recipes/mealie#4219] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

I am Portuguese but a significant part of the recipes I have in Mealie are in English. While I need English recipes to remain in English and Portuguese recipes to remain in Portuguese,  I would still like to treat the ingredients in an uniform way.

Because I generated master data in Portuguese, the zucchini ingredient is "curgete". If I want "zucchini" to be recognized as "curgete", I must manually add it to its pseudonyms. But there are 206 ingredients in my list.

### Please provide a concise description of the feature that would resolve your issue.

I would like:
- the predefined data to already have a translation of all the ingredients in all the different languages
- I would like Mealie to know that, if the recipe is in English, then it must lookup the name of the ingredients in English. Or, alternatively, look up all the possible names of an ingredient to try to find a match in one of the languages.

### Please consider and list out some caveats or tradeoffs made in your design decision

This will require a more complex master data maintenance and will require a new attribute "language" to be added to each recipe. This may be relatively complex. Otherwise I don't see any downside in terms of usability.

I wouldn happily help enrich the translated list of ingredients.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by nalllen** [comment]
Just want to voice my support for this cause it might be a straight up deal breaker for me.

I just installed mealie today and the absolute first thing i noticed when creating my first recipe was this issue with predefined foods/ingredients names all being in English without multi lang support, even changing the application language keeps the ingredients in English.

The issue i have with this is that i would like to add recipes in the original language it comes with, but that would end up with me having to create duplicate ingredients and that in turn would end up being quite the nuisance with multiple entries when searching.

If i want to find every recipe with Apple i would have to check Apple, Äpple, Eple and maçã etc.

If my instructions are in Norwegian i want to be able to just click language-> Norwegian in the recipe edit windows and then have the ingredients show up in the right language.

**Comment by michael-genson** [comment]
> predefined foods/ingredients names all being in English without multi lang support

Seed data is translatable, it sounds like you used the English seed data. You can choose the language before seeding:
![image](https://github.com/user-attachments/assets/e18a6fb5-9bcc-4341-9128-9d9433898c5f)

**Reply by nalllen** [reply]
Glad that this is a option but doesn't really solve the issue since i actively use English recipes and non English ones.

The easiest solution would be to just deal with it and stay with one set of seed data no mater what language the recipe is in,
alternatively seed multiple languages which for me would be 800+ entries in Food Data
both have their own sets of downsides and possible frustrations.

The most straightforward although might have its own set of technical challenges could be to have seed data combine like this
![image](https://github.com/user-attachments/assets/3a8abcef-e8a1-4715-9894-fb41c339193c)

then have a toggle to switch ingredient language, or heck even ai detect it for recipes.

On another note, spend some more timing using mealie and im really liking it ^^

**Comment by jonathan-morton** [comment]
I would like to +1 this. My SO is Japanese, can read English but prefers Japanese. I am the opposite. Some kind of field for adding translations would be great. There is already the field for aliases. Maybe a manage translations as well

**Comment by nununo** [comment]
Mealie ingredients already implement alias. To be honest I didn't understand how these work because they didn't work as expected when I tried searching for an ingredient alias.

Anyway, regardless of calling _alias_, _synonyms_ or something else, the most important is to have sets of words which can be used interchangeably in searches.

Also, ingredients can already be imported for selected languages. We should have a way to correlate the ingredients in different languages. We should treat this as normal translations: we could use the English word as ID and then just maintain an i18n resource file for each language? These could be crowdsourced and then, once imported, they could be further fine tuned. But always keeping each ingredient identified with a unique ID shared by all languages.

**Comment by michael-genson** [comment]
Aliases are currently only used by the ingredient parser. e.g. if a recipe calls for "10 green onions" and "green onions" is an alias of "scallions", then the parser will parse "10 green onions" as "10 scallions". There is room to expand their use, though.

**Reply by Ronaldvr** [reply]
OK! I had to search several times to get here to find the actual use of aliases it is explained nowhere in the documentation (and yet a feature that shows quite prominently in the UI when making or adjusting foods)

Anyhow: In the Netherlands several ingredients used in Indonesian cooking have 2 or more names: 1 Indonesian term and 1 dutch, for instance Ginger is named Djahé as well as 'Gember' (There is even a site dedicated to those : https://www.aziatische-ingredienten.nl/ 

And I expected more or less when I create a shopping list to have the aliases show up (since in Asian stores you get the Asian name for instance)

Furthermore: where is the remark field used for? That is not documented as well.

**Reply by ovizii** [reply]
> Aliases are currently only used by the ingredient parser. e.g. if a recipe calls for "10 green onions" and "green onions" is an alias of "scallions", then the parser will parse "10 green onions" as "10 scallions". There is room to expand their use, though.

So the correlation between an ingredient and its alias can only be made manually while parsing and there is no automatic link between different seed data?

**Reply by michael-genson** [reply]
Currently aliases are completely absent from seed data. You can add aliases manually via the data management page, though (only recently could you add them through parsing)

**Reply by laurens-moonens** [reply]
Is this a feature that is planned in the future? I'm running into the same issue with English and Dutch. I use some Dutch recipes and some English recipes. So I'd like to seed my data with BOTH English and Dutch, so both languages can be parsed easily, without having to manually create aliases for everything.

**Reply by laurens-moonens** [reply]
Testing a bit more, the current behavior is quite weird. I have my data seeded with English. When I add Dutch seeding, nothing gets added. But when I remove 1 English ingredient, and seed again using Dutch, then that removed ingredient IS added back in Dutch. So Mealie DOES seem to track some correlation between the Dutch and English ingredients when seeding, but not when parsing.

**Reply by MikiiTakagi** [reply]
I'm also running the same issue with English and French. I'm also looking into having multi-language recipe but that another concern as I could simply have a recipe for each language. Lot of feature are linked to ingredient so this is the main concern for me currently.

**Comment by Pat4fix** [comment]
+1 for this. It's kind of a deal-breaker for me because my own recipes are in German, so my shopping list should be in German—but every recipe I find on the Internet is in English. So that's a problem. And for some reason, I also can’t seed with a second language. Is that normal?

**Comment by philchamp** [comment]
+1 for this also. I have recipes in both French and English. My base ingredients are in english and when I try to seed french canadian ingredients, it only adds 7 ingredients if I remember correctly and almost (if not) all of them are mushrooms for some reasons.
